test/rules/no-improper-constructor-param-usage/test.ts.lint

```diff
@@ -0,0 +1,48 @@
+class FailingExample1 {
+    constructor(fruit: string, public num: number, private animal: string) {
+        this.num = 10;
+        num = 5;
+        ~~~ [The constructor parameter 'num' should only be accessed with the 'this' keyword: 'this.num']
+
+        animal = "aardvark";
+        ~~~~~~ [The constructor parameter 'animal' should only be accessed with the 'this' keyword: 'this.animal']
+        this.animal = "skunk";
+
+        fruit = "banana";
+        ~~~~~ [The constructor parameter 'fruit' should only be accessed with the 'this' keyword: 'this.fruit']
```

Comment:
Ah good point.  i suppose this rule should only check params that have a `public` or `private` modifier on it, correct?